tests/smart/modules/test_fan.py

```diff
@@ -77,8 +78,37 @@ async def test_fan_module(dev: SmartDevice, mocker: MockerFixture):
     await dev.update()
     assert not device.is_on
 
+    fan_speed_level_feature = fan._module_features["fan_speed_level"]
+    max_level = fan_speed_level_feature.maximum_value
+    min_level = fan_speed_level_feature.minimum_value
     with pytest.raises(ValueError, match="Invalid level"):
-        await fan.set_fan_speed_level(-1)
+        await fan.set_fan_speed_level(min_level - 1)
 
     with pytest.raises(ValueError, match="Invalid level"):
-        await fan.set_fan_speed_level(5)
+        await fan.set_fan_speed_level(max_level - 5)
+
+
+@fan
+async def test_fan_features(dev: SmartDevice, mocker: MockerFixture):
+    """Test fan speed on device interface."""
+    assert isinstance(dev, SmartDevice)
+    fan = next(get_parent_and_child_modules(dev, Module.Fan))
+    assert fan
+    expected_feature = fan._module_features["fan_speed_level"]
+    fan_speed_level_feature = fan.get_feature(Fan.set_fan_speed_level)
+    assert expected_feature == fan_speed_level_feature
+    fan_speed_level_feature = fan.get_feature(fan.set_fan_speed_level)
+    assert expected_feature == fan_speed_level_feature
+    fan_speed_level_feature = fan.get_feature(Fan.fan_speed_level)
+    assert expected_feature == fan_speed_level_feature
+    fan_speed_level_feature = fan.get_feature("fan_speed_level")
+    assert expected_feature == fan_speed_level_feature
+    assert fan.has_feature(Fan.fan_speed_level)
```

Comment:
```suggestion
    expected_feature = fan._module_features["fan_speed_level"]

    fan_speed_level_feature = fan.get_feature(Fan.set_fan_speed_level)
    assert expected_feature == fan_speed_level_feature

    fan_speed_level_feature = fan.get_feature(fan.set_fan_speed_level)
    assert expected_feature == fan_speed_level_feature

    fan_speed_level_feature = fan.get_feature(Fan.fan_speed_level)
    assert expected_feature == fan_speed_level_feature

    fan_speed_level_feature = fan.get_feature("fan_speed_level")
    assert expected_feature == fan_speed_level_feature

    assert fan.has_feature(Fan.fan_speed_level)
```
No need to save white spaces here to make the code more easier to read :-)